src/misc/support.cpp

```diff
@@ -36,7 +37,21 @@
 #include "debug.h"
 #include "video.h"
 
-std::string get_basename(const std::string& filename) {
+char int_to_char(int val)
+{
+	// Confirm the incoming value is either within the existing CHAR range
+	// (-128..127) or, if it's coming from a DOS API call, the unsigned
```

Comment:
Nitpick: `char` is either signed or unsigned, depending on *compiler configuration* (can be enforced via compiler flags). Some OSes prefer signed chars by default, other OSes have unsigned ones.

The code is correct (using CHAR_MIN); the comment might be misleading.

Reply:
Thanks for this distinction @dreamer ... no wonder C/C++ has got itself in such a security mess with raw string processing.

Like a big oak tree, the the C++ standards committee is cranking out glorious new and safe constructions in C++20 as fresh new green leaves, but the foundation and trunk of the tree is rotten with this nonsense!  

I'm baffled why they haven't deprecated all their unsafe and ambiguous behavior messes incrementally across standards.  Say, first as an INFO in C++11, then as a WARN in C++14, and then REMOVED in C++17. It could be a proper and safe language by now.. but instead C++ with be _**rusted**_ away ;-)

Reply:
Fixed with:

``` C++
	// To handle inbound values cast from unsigned chars, permit a slightly
	// wider range to avoid triggering the assert when processing international
	// ASCII values between 128 and 255.
	assert(val >= CHAR_MIN && val <= UCHAR_MAX);
	return static_cast<char>(val);
```